File: problems_026to050/problem_036/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch
from toolbox import is_palindrome


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0
        for num in range(1, 1000000):
            if is_palindrome(str(num)):
                binary = format(num, 'b')
                if is_palindrome(binary):
                    answer += num
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_036/solution.js
const { isPalindrome } = require('../../lib/toolbox');

const solution = () => {
  let answer = 0;

  for (let num = 1; num < 1000000; num += 1) {
    if (isPalindrome(String(num))) {
      const binary = num.toString(2);

      if (isPalindrome(binary)) {
        answer += num;
      }
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
